pkg/server/vault.go

```diff
@@ -0,0 +1,405 @@
+func createVaultClient(vaultAddr, vaultToken string, timeout time.Duration) (*api.Client, *VaultError) {
+	config := api.DefaultConfig()
+	config.Address = vaultAddr
+	config.HttpClient.Timeout = timeout * time.Second
+	client, err := api.NewClient(config)
+	if err != nil {
+		return nil, &VaultError{
+			
+			Message: fmt.Sprintf(VaultErrorCreatClient, err),
+		}
+	}
+	client.SetToken(vaultToken)
+	return client, nil
+}
+
+// enableKVSecretsEngine enables the KV (Key-Value) secrets engine (version 2) at the specified path in Vault.
+//
+// Parameters:
+// - client: A pointer to the Vault API client.
+// - path: The mount path where the KV secrets engine should be enabled.
+//
+// Returns:
+// - error: An error if the operation fails; otherwise, nil.
+//
+// This function sends a request to the Vault server to mount the KV v2 secrets engine
+// at the specified path. It constructs the request body, makes an HTTP request, and
+// validates the response status.
+func enableKVSecretsEngine(client *api.Client, path string) error {
+	// Prepare the request body for enabling kv-v2 secrets engine
+	mountRequest := map[string]interface{}{
+		"type":        "kv",
+		"options":     map[string]interface{}{"version": "2"}, // specify kv-v2 version
+		"description": "KV secrets engine version 2",
+	}
+
+	// Marshal the request body into JSON
+	mountRequestBody, err := json.Marshal(mountRequest)
+	if err != nil {
+		return fmt.Errorf("failed to marshal mount request: %v", err)
+	}
+
+	// Use the Vault client to send an HTTP request to the sys/mounts endpoint
+	req := client.NewRequest("POST", fmt.Sprintf("/v1/sys/mounts/%s", path))
+
+	// Set the request body
+	req.Body = bytes.NewBuffer(mountRequestBody)
+
+	// Send the request to enable the KV engine at the given path
+	resp, err := client.RawRequest(req)
+	if err != nil {
+		return fmt.Errorf("failed to enable KV engine at path %s: %v", path, err)
+	}
+
+	if resp.StatusCode != 204 {
+		return fmt.Errorf("failed to enable KV engine, unexpected status code: %v", resp.StatusCode)
+	}
+
+	fmt.Printf("Successfully enabled KV version 2 at path: %s\n", path)
+	return nil
+}
+
+// StartClient starts a local Vault server in development mode.
+//
+// This function is only called when using a locally installed Vault. It performs the following steps:
+// 1. Kills any existing Vault process.
+// 2. Starts a new Vault server in development mode with a predefined root token.
+// 3. Sets the VAULT_ADDR environment variable based on the operating system.
+// 4. Enables the KV secrets engine at the "secret" path.
+//
+// Returns:
+// - *VaultError: An error object if the Vault process fails to start or if setting the environment variable fails.
+func (v *VaultClient) StartClient() *VaultError {
+	// Kill existing Vault process
+	v.CloseClient()
+	// Start Vault in dev mode
+	vaultCmd = exec.Command("vault", "server", "-dev", "-dev-root-token-id="+v.client.Token())
```

Comment:
Let's use docker-compose for this, or assume vault has been started by the outside. We don't need to spawn dev dependencies inside Go code...

Reply:
This part is only for local installed Vault client for test.
Can we remove this? i am not sure i need move this func to other files.

Reply:
I removed code in new commit